src/test/java/org/cloudfoundry/promregator/PromregatorApplicationTest.java

```diff
@@ -19,14 +20,9 @@
 import org.springframework.test.context.TestPropertySource;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
+@SpringBootTest(classes = PromregatorApplication.class)
 @ExtendWith(SpringExtension.class)
-@BootstrapWith(value=SpringBootTestContextBootstrapper.class)
-@ComponentScan(excludeFilters = { @Filter(type = FilterType.CUSTOM, classes = { TypeExcludeFilter.class }),
-		@Filter(type = FilterType.CUSTOM, classes = { AutoConfigurationExcludeFilter.class })
-		// NB: TestableMetricsEndpoint would break here everything
-})
 @TestPropertySource(locations="default.properties")
-@DirtiesContext(classMode=ClassMode.AFTER_CLASS)
```

Comment:
This looks dangerous to me: Won't it leave the context dirty behind?
We have numerous cases where this could lead to issues...

Reply:
Only if we know we are dirtying the context. In this test it clearly isn't since it just starts up the context and verifies it will load.

Spring likes to reuse contexts for performance reasons, so I'm fine marking the context as dirty, but only if we are actually changing stuff. Our only 'stateful' beans are the caches, so we should be safe here (and likely even with the caches).

Also keep in mind a context has to be 100% identical (same beans and properties and profiles) to be reused.